fastembed/text/onnx_embedding.py

```diff
@@ -254,6 +269,11 @@ def init_embedding(
         self,
         model_name: str,
         cache_dir: str,
+        device_id: Optional[int] = None,
         **kwargs,
     ) -> OnnxTextEmbedding:
+        providers = kwargs.get("providers", None)
+        if device_id is not None and providers and "CUDAExecutionProvider" in providers:
+            kwargs["providers"] = [("CUDAExecutionProvider", {"device_id": device_id})]
```

Comment:
We either allow the user to specify exact providers they want, or we completely abstract providers away and do everything for them.

But this approach takes the worst parts of both: we require users to "know" about onnx and it's guts, force them to specify exact names of those drivers ONNX uses internally, but once they do it, we don't respect any custom parameters they provide. In this case we throw those parameters away completely. This will be very frustrating for those who actually want advanced usage. 

Again, this is a very reason why we discussed `cuda=True/False` parameter, please explain why it is not used?